### DISCUSSION TITLE
Bug or am I missing a setting?

### DISCUSSION THREAD

**Discussion by mixlplix** [oqtane/oqtane.framework#1775] [General]
I have a module that lists products and then has an edit button to edit each product.  I moved the module into the "Top Full Width" pane in the Theme and the Index page shows up in that pane at full width, but when I navigate to Edit.razor, the page shows up in the "Top 100%" pane for some reason.  It's the edit page that I really need to show up in the full width page.  I am using this setting 

    public override bool UseAdminContainer => false;

because I don't want the edit page to show up in a modal, but I don't understand why the edit/add components do not render in the same Pane that I have added the module to?

**Comment by mixlplix** [comment]
Just to follow up on this.  I also created a Container that was full width and you can see from the screenshots, the difference between the two,

![image](https://user-images.githubusercontent.com/1314264/141017270-0dcddc01-577b-41c9-a142-c40353296e8b.png)

and the edit page
![image](https://user-images.githubusercontent.com/1314264/141017309-34faded8-993f-4733-90cc-19bd78a7c61e.png)

**Reply by leigh-pointer** [reply]
@mixlplix I would need to see the container markup but my first opinion is that you are using class="container" and not class="container-fluid" you and also make the container even tighter by adding g-0 to the class.

**Comment by mixlplix** [comment]
Nope, that’s not it, here’s my Container code:

 

 

@namespace Ipp.Theme

@inherits ContainerBase

 

<div class="container-fluid">

    Container Begins Here

    <div class="px-4">

        <div class="d-flex flex-nowrap">

            <ModuleActions /><h2><ModuleTitle /></h2>

        </div>

        <hr class="app-rule" />

    </div>

    <div class="px-4">

        <div class="container-fluid">

            <ModuleInstance />

        </div>

    </div>

</div>

 

@code {

    public override string Name => "ContainerFullWidth";

}

 

 

I placed the works “Container Begins Here” on purpose so I could inspect the html around it, and there is another div class=“container” above in the DOM that is constricted it.  I also put some text in the admin container to see if it is being displayed there, and it appears that is the case.  Look at the attached image with the developer console on…

 



 

And here’s the partial Theme:

 



 

</doug>

**Comment by sbwalker** [comment]
@mixlplix this is a great question...

The UseAdminContainer property defines whether or not the framework should use the "admin container" which provides the modal popup behavior for "admin" components. Note that this behavior can also be configured by setting the Default Admin Container option in Site Settings ( this option has some advantages as it allows you to define a specific custom container as well - however it cannot be set programmatically ). 

Either way, neither of these options have any relation to "Panes" - which is the ultimate location where a module is injected into the theme. The default behavior for "admin" components is that they are always injected into the "Admin" pane of the theme. So even if the "index" component is configured to be injected into the "Top Full Width" pane of the page... the related "admin" components for the module will be injected into the "Admin" pane. 

This is "by design". The background to this area is that when Oqtane was created there was an effort to simplify some of the more cumbersome UI concepts which exist in DNN. One area in particular was the concept in DNN where you needed to define both a "Default Theme" and "Admin Theme" for your site - a concept which created extra work for theme designers and extra complexity in the framework and user experience, for a use case which was seldomly utilized. Oqtane eliminated the need for a dedicated "Admin Theme" and chose instead to leverage a single theme for both purposes - with the ability to influence behavior via containers. 

So back to your original question... the quickest solution is for you to create a custom theme that has an "Admin" pane that is full width. 

However there is also a philosophical question to consider related to the default framework behavior. Currently it relies on the Admin pane. However, there is no reason why it could not leverage the pane specification for the Module. So if the module was configured to display in the "Top Full Width" pane, its admin component could certainly use this guidance as well - as it is guaranteed that the pane exists in the theme (which was not the case in the DNN scenario where there were 2 independent themes). If the framework was changed to use this approach, it would not affect the default behavior as the modal popup admin container can be injected in any pane and still behave the same. However there still may be cases where a module is designed to display its Index content in a very size constrained pane - but the administrative features need to be displayed in a full size pane. In this case the developer would need to use a modal popup container as there is no way for them to provide an alternate pane specification. Which obviously leads to the idea of introducing another configuration option - however the problem is that themes and modules are designed to be completely independent from one another without any dependencies - so even if there was an option such as AdminPaneName which a developer could set in their module, there is no guarantee the theme being used actually contains the pane specified. So this is not a viable approach. 

So I guess the only real question is whether the default framework behavior should be modified to use the Module pane specification rather than the current hardcoded "Admin" pane? I am open to feedback.

**Reply by mixlplix** [reply]
Thanks for the background info on this @sbwalker.  I would vote for having the "admin" pages show up in the same container, if someone doesn't want that then they use the modal option.  For us, the entire site is an admin site, so we would prefer that to be consistent.  Having said that, I tried doing what you said, and moving the admin pane into the full width section, and it still does not work.  I have identified the areas with text that I placed in the Theme.  See below first the screenshot of the page not in full width mode, and then my Theme.
![image](https://user-images.githubusercontent.com/1314264/141373599-604bc077-f512-4d93-9bf8-0ef9df993584.png)

Here is the Theme
![image](https://user-images.githubusercontent.com/1314264/141373659-cbb8d616-70a4-4c0a-b46d-c10a9e76a61b.png)

Still looks like something in code is adding the container where it shouldn't be...

**Reply by mixlplix** [reply]
Wait, I figured it out.  We are using Blazorise, and the <Container> class we were using, even though I was passing a class was just appending the class "container-fluid" to container.  This one is solved,

**Reply by sbwalker** [reply]
@mixlplix there is one other aspect to the current behavior for which I would like to get some feedback. Currently when you are in an "admin" context ( ie. you have invoked an action on a module which involves the display of a different component for that module ), none of the other modules which may exist on that page are displayed - only the module upon which you took the action is rendered. Under default circumstances the module is displayed in a modal popup container so you may not even realize that the other modules are not rendered. However in the scenario where you are not using a modal popup container it is very obvious that only a single module component is displayed. If the behavior is changed so that the module component is rendered in the Pane defined by the module and not hardcoded to the Admin pane ( as discussed above ) then I am wondering if it also makes sense to also render the other modules which may exist on the page? This might make the "in-line" page management experience more fluid in those scenarios where modal popups are not being used. Or it might be a distraction for users as it would no longer keep them focused on the module they were interacting with. Thoughts?

**Reply by mixlplix** [reply]
Interesting.  We pretty much adhere to a one module one page scenario, so this is a scenario I would not have contemplated.  If I did have multiple modules on a page, those modules would have been explicitly placed in a pane, so I would expect the admin context for each of those modules to render in the same pane unless I decided that a modal was better.  I am guessing that if I did have more than one module on a page, I would likely always invoke the modal for editing purposes any way.  The most compelling reason for not using a modal is that the amount of data you are editing is too large to reasonably fit in a modal, and in those cases it would be unlikely that I would have more than one module on that page.  Hope that helps.
